internal/daemonset/daemonset.go

```diff
@@ -53,10 +54,20 @@ func NewCreator(client client.Client, kernelLabel string, scheme *runtime.Scheme
 	}
 }
 
-func (dc *daemonSetGenerator) GarbageCollect(ctx context.Context, existingDS []appsv1.DaemonSet, validKernels sets.Set[string]) ([]string, error) {
+func (dc *daemonSetGenerator) GarbageCollect(ctx context.Context, mod *kmmv1beta1.Module, existingDS []appsv1.DaemonSet, validKernels sets.Set[string]) ([]string, error) {
 	deleted := make([]string, 0)
 
+	versionLabel := utils.GetModuleVersionLabelName(mod.Namespace, mod.Name)
 	for _, ds := range existingDS {
+		// delete inactive device-plugin or module-loader DS with former label
+		if ds.Labels[versionLabel] != mod.Spec.ModuleLoader.Container.Version && ds.Status.NumberAvailable == 0 {
```

Comment:
Should we instead check the number of desired pods?

Reply:
both are ok, but "desired" is better and cleaner.
Fixed